docs/spec_draft.md

```diff
@@ -1061,6 +1099,74 @@ tensor and produces a `result` tensor.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.custom_call
+
+### Semantics
+
+Invokes exteral code with `inputs` as arguments and produces `results`. The
+exact mechanism is backend-specific. The inputs are described as follows:
+  * `inputs` are passed as agruments to the external API.
+  * `call_target_name` may used as an identifier for the external code.
+  * `has_side_effect` is true if the custom call has side-effects.
+  * `backend_config` can be used to encode additional information.
+  * `api_version` specifies the version of the custom call API.
+  * `called_computations` used to apply functions within the scope of the parent
+    module.
+  * `operand_layouts` specifies the layout of the operand.
+  * `result_layouts` specifies the layout of the result.
+  * `output_operand_aliases` asiases attributes for results and operands.
+
+### Inputs
+
+| Name                     | Type                                                                |
+|--------------------------|---------------------------------------------------------------------|
+| `inputs`                 | variadic number of tensors of any supported type, tokens, or tuples |
+| `call_target_name`       | `string`                                                            |
+| `has_side_effect`        | `i1`                                                                |
+| `backend_config`         | `string`                                                            |
+| `api_version`            | enum of `API_VERSION_ORIGINAL`, `API_VERSION_STATUS_RETURNING`,     |
+|                          | and `API_VERSION_STATUS_RETURNING_UNIFIED`                          |
+| `called_computations`    | array of `function` references                                      |
+| `operand_layouts`        | array of 1-dimensional tensor constants of type `index`             |
+| `result_layouts`         | array of 1-dimensional tensor constants of type `index`             |
```

Comment:
Just talked to Sandeep about something similar in `all_reduce`. Instead of "array of", we can use "variadic number of", so that we don't have to introduce a notion of arrays in the spec.